Source/control.cpp

```diff
@@ -741,17 +753,19 @@ void DrawCtrlPan()
 	int i;
 
 	for (i = 0; i < 6; i++) {
-		if (!panbtn[i])
-			DrawPanelBox(PanBtnPos[i][0], PanBtnPos[i][1] - 336, 71, 20, PanBtnPos[i][0] + SCREEN_X, PanBtnPos[i][1] + SCREEN_Y);
+		if (panbtn[i])
+			CelDraw(PanBtnPos[i][0] + SCREEN_X, PanBtnPos[i][1] + SCREEN_Y + PanBtnPos[i][3] - 1, pPanelButtons, i + 1, PanBtnPos[i][2]);
 		else
-			CelDraw(PanBtnPos[i][0] + SCREEN_X, PanBtnPos[i][1] + SCREEN_Y + 18, pPanelButtons, i + 1, 71);
+			if (SCREEN_WIDTH != BASE_WIDTH || SCREEN_HEIGHT != BASE_HEIGHT)
+				//This line makes ASAN crash and doesn't work properly on some resolutions but the panel is always being redrawn on custom resolutions, so we don't need it there
+				DrawPanelBox(PanBtnPos[i][0], PanBtnPos[i][1] - 336, PanBtnPos[i][2], PanBtnPos[i][3] + 1, PanBtnPos[i][0] + SCREEN_X, PanBtnPos[i][1] + SCREEN_Y);

```

Comment:
The ASAN could be fixed by calling DrawPanelBox that way
~~~~~
DrawPanelBox(PanBtnPos[i][0] - WIDTH_DIFF_2, PanBtnPos[i][1] - HEIGHT_DIFF- 336, PanBtnPos[i][2],
PanBtnPos[i][3] + 1,
PanBtnPos[i][0] + SCREEN_X - WIDTH_DIFF_2,
PanBtnPos[i][1] + SCREEN_Y - HEIGHT_DIFF);
~~~~~

Reply:
but are the buttons drawn in correct places with this?